### PR TITLE
Nrel#166 automatically download road network if doesnt exist

### PR DESCRIPTION
**Pull request by clintonsteiner**

### CODE REVIEW

File: nrel/hive/initialization/initialize_simulation.py [hunk 1]
```diff
@@ -240,14 +240,31 @@ def osm_init_function(
...
+        dataframe = geopandas.read_file(config.input_config.geofence_file)
+        polygon_union = dataframe["geometry"].unary_union
+
+        road_network = OSMRoadNetwork.from_polygon(
+            sim_h3_resolution=config.sim.sim_h3_resolution,
+            default_speed_kmph=config.network.default_speed_kmph,
+            polygon=polygon_union,
+        )
+    else:
+        raise IOError("Must supply a road network file when using the osm_network")
```
Comment (robfitzgerald) [inline nrel/hive/initialization/initialize_simulation.py:267]: this is the error message for omission of either road_network or geofence_file. the error should reflect this, perhaps via `raise IOError("Must supply either a road network or geofence file when using the osm_network")`

File: tests/test_initialize_simulation.py [hunk 2]
```diff
@@ -79,3 +84,33 @@ def test_initialize_simulation_with_sampling(self):
         self.assertEqual(len(sim.vehicles), 20, "should have loaded 20 vehicles")
         self.assertEqual(len(sim.stations), 4, "should have loaded 4 stations")
         self.assertEqual(len(sim.bases), 2, "should have loaded 2 bases")
+
+    def test_initialize_simulation_load_from_geofence_file(self):
```
Comment (robfitzgerald) [inline tests/test_initialize_simulation.py:88]: i appreciate the idea of testing this, but, this implies every time we test hive that we are making a call to OpenStreetMaps. we probably shouldn't do this unless we can cache the query result locally for osmnx, which may not be worth our time to figure out. perhaps we flag this test with `@unittest.skip("makes API call to OpenStreetMaps via osmnx")`?

File: nrel/hive/model/roadnetwork/osm/osm_builders.py [hunk 3]
```diff
@@ -13,6 +13,7 @@ def osm_graph_from_polygon(polygon):
         ) from e
 
     ox.settings.all_oneway = True
+    ox.settings.use_cache = False
```
Comment (robfitzgerald) [inline nrel/hive/model/roadnetwork/osm/osm_builders.py:16]: why not use cache?

**Review by robfitzgerald — CHANGES_REQUESTED** [review]
solid start and one step closer to a really simple way for people to use HIVE.

i have some requests for fixes and one question as well.

:bee:

File: tests/test_initialize_simulation.py (hunk 2 above)
Comment (clintonsteiner) [reply tests/test_initialize_simulation.py:88]: Agreed, added

File: nrel/hive/initialization/initialize_simulation.py (hunk 1 above)
Comment (clintonsteiner) [reply nrel/hive/initialization/initialize_simulation.py:267]: Fixed

File: nrel/hive/model/roadnetwork/osm/osm_builders.py (hunk 3 above)
Comment (nreinicke) [reply nrel/hive/model/roadnetwork/osm/osm_builders.py:16]: I wonder if we could set the cache folder instead to cache the road network results in the scenario folder using the `cache_folder` setting from [here](https://osmnx.readthedocs.io/en/stable/osmnx.html?highlight=config#module-osmnx.settings)?

Comment (clintonsteiner) [reply nrel/hive/model/roadnetwork/osm/osm_builders.py]: Talked with Nick and we got this updated, unsure if I love the implementation though. Open to suggestions

**Review by robfitzgerald — APPROVED** [review]
lgtm, thanks for adding this!

**Pull request merged**